server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java

```diff
@@ -150,16 +157,14 @@ public void buildInterrupted(@NotNull SRunningBuild build) {
     }
 
     private void buildFinishedOrInterrupted (SRunningBuild build) {
-        String buildId = getBuildId(build);
-        String buildName = getBuildName(build);
-        Loggers.SERVER.debug("OTEL_PLUGIN: Build finished method triggered for " + buildId);
+        Loggers.SERVER.debug("OTEL_PLUGIN: Build finished method triggered for " + getBuildId(build));
 
         BuildStatistics buildStatistics = build.getBuildStatistics(
                 BuildStatisticsOptions.ALL_TESTS_NO_DETAILS);
 
-        if(this.otelHelper.getSpan(buildId) != null) {
-            Span span = this.otelHelper.getSpan(buildId);
-            Loggers.SERVER.debug("OTEL_PLUGIN: Build finished and span found for " + buildName);
+        if(this.otelHelper.getSpan(getBuildId(build)) != null) {
```

Comment:
I think I preferred this a a variable, rather than calling the method 3 different times?